Source/DafnyTests/DafnyTestCase.cs

```diff
@@ -0,0 +1,417 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.FileProviders;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using XUnitExtensions;
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+using YamlDotNet.Serialization.ObjectFactories;
+
+namespace DafnyTests {
+  
+  public class DafnyTestSpec : IEnumerable<DafnyTestCase> {
+
+    private static DirectoryInfo OUTPUT_ROOT = new DirectoryInfo(Directory.GetCurrentDirectory());
+    
+    // TODO-RS: This is an ugly method of locating the project root.
+    // The proper fix is to run entirely out of the output directory,
+    // and the projects are at least partially configured to make that possible,
+    // but it's not quite working yet.
+    private static string DAFNY_ROOT = 
+      OUTPUT_ROOT.Parent.Parent.Parent.Parent.Parent.FullName;
+
+    public static readonly string TEST_ROOT = Path.Combine(DAFNY_ROOT, "Test") + Path.DirectorySeparatorChar;
+    public static readonly string COMP_DIR = Path.Combine(TEST_ROOT, "comp") + Path.DirectorySeparatorChar;
```

Comment:
np: I wish all of these folders followed similar capitalization. Def don't expect this to be fixed.

Reply:
Not with that attitude :) Definitely not in THIS PR though!